notebooks/2.single-cell-compound-prio/4.selecting_hits.py

```diff
@@ -0,0 +1,218 @@
+        contains "failing", returns "DMSO-failing". If the treatment is "DMSO" and
+        the cell type does not contain "failing", returns "DMSO-healthy". Otherwise,
+        returns the original treatment name.
+    """
+    if treatment_name in "DMSO":
+        if "failing" in cell_type:
+            return "DMSO-failing"
+        else:
+            return "DMSO-healthy"
+    else:
+        return treatment_name
+
+
+# setting up paths
+
+# In[3]:
+
+
+# setting results directory
+results_dir = pathlib.Path("./results").resolve(strict=True)
+
+# setting kl divergence score path
+kl_divergence_score_path = (results_dir / "metrics/_kl_divergence_scores.csv").resolve(
+    strict=True
+)
+
+# setting treated and control paths
+trt_pathway_path = pathlib.Path(
+    "../data/metadata/original_platemaps/pathways_platemap.csv"
+).resolve(strict=True)
+
+# create a results folder known as "hits"
+hits_dir = (results_dir / "hits").resolve()
+hits_dir.mkdir(exist_ok=True)
+
+
+# Loading in profiles and metadata
+
+# In[4]:
+
+
+# loading treatment pathway data
+pathway_df = pd.read_csv(trt_pathway_path)
+
+# loading in the kl divergence scores
+scores_df = pd.read_csv(kl_divergence_score_path)
+
+print(scores_df.shape)
+scores_df.head()
+
+
+# ## Finding potential hits
+#
+# Here, we load the scores to identify compounds that demonstrate potential reversal effects. Using the loaded scores, we rank the compounds based first on their on-morphology scores and then on their off-morphology scores. These scores were generated using the positive control (healthy CF cells + DMSO), where smaller scores indicate a stronger potential for reversal. Specifically, lower on-morphology scores suggest that the treated cells closely resemble the positive control, while lower off-morphology scores indicate minimal off-target effects.
+
+# In[5]:
+
+
+# identify_hits
+scores_df = scores_df
+method = "mean"
+strata = ["treatment_name"]
+
+scores = []
+for strata_name, score_df in scores_df.groupby(strata):
+    # if the method is 'mean', then we take the mean of both on and off scores
+    if method == "mean":
+        off_morph_score = score_df["off_kl_divergence"].mean()
+        on_morph_score = score_df["on_kl_divergence"].mean()
+        comb_morph_score = score_df["combined_score"].mean()
+    # if the method is 'max', then we take the max of both on and off scores
+    elif method == "max":
+        off_morph_score = score_df["off_morph_score"].max()
+        on_morph_score = score_df["on_morph_score"].max()
+        comb_morph_score = score_df["combined_score"].max()
+    # if the method is 'median', then we take the median of both on and off scores
+    elif method == "median":
+        off_morph_score = score_df["off_morph_score"].median()
+        on_morph_score = score_df["on_morph_score"].median()
+        comb_morph_score = score_df["combined_score"].median()
+    else:
+        raise ValueError(f"Method {method} not recognized")
```

Comment:
I see you can perform multiple methods to do this task, why not do them all and have plots for each method of combining the on and off scores?

Reply:
Performed all and created figures!